internal/guest/runtime/hcsv2/uvm_state.go

```diff
@@ -0,0 +1,96 @@
+//go:build linux
+// +build linux
+
+package hcsv2
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"sync"
+)
+
+type rwDevice struct {
+	mountPath  string
+	sourcePath string
+	encrypted  bool
+	overlays   map[string]struct{}
+}
+
+type hostMounts struct {
+	stateMutex sync.RWMutex
+
+	// Holds information about read-write devices, which can be encrypted and
+	// contain overlay fs upper/work directory mounts.
+	readWriteMounts map[string]*rwDevice
```

Comment:
I thought about using it in the beginning to avoid adding a mutex, but I think we have to make sure we're not mounting anything on top of an existing mount. One could replace the existing scratch that way.